### PR TITLE
Validated input text fields to enter only blank spaces

### PR DESCRIPTION
**Pull request by RonakParmar**
### Description (*)

Advanced Search allows search by entering only blank spaces in all the fields of the form and it will generate blank search result. which is not correct behavior of search result.

### Fixed Issues (if relevant)

1. Fixes magento/magento2#23055: Advanced Search Allows Search using Blank Spaces 

### Manual testing scenarios (*)

1. Click on Advanced Search link in the footer
2. Enter only blank spaces in any/all fields
3. Click on Search button
    

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=23059) <br/>All committers have signed the CLA.

File: app/code/Magento/CatalogSearch/view/frontend/templates/advanced/form.phtml
Comment (orlangur) [inline app/code/Magento/CatalogSearch/view/frontend/templates/advanced/form.phtml]: Hi @RonakParmar. Such validation is not correct, you should check that entered string is not empty after it is trimmed instead.

What's the problem with currently observed result?

**Review by orlangur — CHANGES_REQUESTED** [review]

File: app/code/Magento/CatalogSearch/view/frontend/templates/advanced/form.phtml
Comment (RonakParmar) [reply app/code/Magento/CatalogSearch/view/frontend/templates/advanced/form.phtml]: @orlangur Currently if you enter only blank spaces in any of the fields then it will generate search result of it, which is not correct, So, I have done validation if user enters only spaces and click the Search button it will validate the input field and throw an error to enter valid string. I have fixed it in my local setup, please have a look.
#23055-fixed-issue

**Comment by orlangur** [comment]
@magento give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @orlangur. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @orlangur, here is your [Magento instance](https://i-23059-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-23059-2-3-develop.instances.magento-community.engineering/admin](https://i-23059-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by RonakParmar** [comment]
Hi @orlangur , As you can see, I have created new PR to fix this issue. All checks have been successfully passed except "Static Tests build" test. I have checked the test result but not able to understand how to fix it. Can you please help me to clear the test? 

Thank you.

**Comment by orlangur** [comment]
@RonakParmar it's not related to your changes https://github.com/magento/magento2/blob/a2ca17055e9282ccf50068bb9a760c491abbc2da/app/code/Magento/CatalogSearch/Model/Advanced.php#L184

It was trying to say that tags needs to be either aligned identically or not aligned at all.

I'll this functionality a try this week in in the beginning of next.

**Comment by RonakParmar** [comment]
@orlangur Do I need to do anything in this PR to get approval from Maintainer?  Will you please review this PR?
Thank you.

**Comment by RonakParmar** [comment]
@magento-engcom-team Can you please guide me? What next to do in this PR? Who will review this? 
Thank you.

**Review by sidolov — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sidolov, thank you for the review.
[ENGCOM-5622](https://jira.corp.magento.com/browse/ENGCOM-5622) has been created to process this Pull Request
:eight_spoked_asterisk: @sidolov, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by magento-engcom-team** [comment]
@RonakParmar thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

**Comment by engcom-Delta** [comment]
:heavy_check_mark: QA passed

**Pull request merged**
